### DISCUSSION TITLE
EdgeX - Resiliency/Fail Safe Features

### DISCUSSION THREAD

**Discussion by vijay-hcl-epl** [edgexfoundry/community#79] [Q&A] [answered]
Hi all,
We have been using EdgeX for quite sometime. Whenever we propose EdgeX to our clients, they ask us how robust/resilient is EdgeX? How all the microservices handle errors and ensure EdgeX is **ALWAYS AVAILABLE**? So we need to understand whether the fail-safe mechanisms are implemented in EdgeX or not. For example, consider the below scenario:

> 1. If EdgeX Message bus fails for some reason, does it recover without any issues?

Like this any microservice can run into errors. Could you please share some links or documents to understand how resilient is EdgeX? As a user do we need to do anything to  handle any such errors? 

Please let us know. It will be very helpful.

Thanks,
Vijay.

**Comment by bnevis-i** [comment, marked as answer]
@vijay-hcl-epl Please see the following writeup:  https://docs.edgexfoundry.org/2.4/general/EdgeX_CN/, particularly, the question "Is EdgeX HA?"

In particular, if you have a need for HA, the community has a process called "Use Case Requirements" where you can describe your requirements, such as what kind of failures that EdgeX needs to be able to recover from, and this document can be the basis of architectural changes to support it.  Also if you have any SLA's that need to be complied with, like recovery time for various services failing, that would be important as well to capture in the UCR.

See [Use Cases and Design Process](https://docs.edgexfoundry.org/2.3/design/Process/)

**Reply by vijay-hcl-epl** [reply]
Thank you @bnevis-i  for the response.

**Reply by bnevis-i** [reply]
Looking forward to information about your use case
